src/language-js/printer-estree.js

```diff
@@ -1518,8 +1518,7 @@ function printPathNoParens(path, options, print, args) {
         //
         // Note that getLast returns null if the array is empty, but
         // we already check for an empty array just above so we are safe
-        const needsForcedTrailingComma =
-          canHaveTrailingComma && lastElem === null;
```

Comment:
I think it'd be better to preserve `canHaveTrailingComma` since it can be changed in the future and I don't think we'll remember to add it back if it's not `true`.